tasking/pulp/tasking/scheduler.py

```diff
@@ -258,7 +258,7 @@ def tick(self):
                           % {'celerybeat_name': self.celerybeat_name})
             ret = self.call_tick(self, self.celerybeat_name)
         except TaskLock.DoesNotExist:
-            TaskLock.objects.get(name=self.celerybeat_name, timestamp__lte=old_timestamp).delete()
+            TaskLock.objects.filter(name=self.celerybeat_name, timestamp__lte=old_timestamp).delete()
```

Comment:
The resource manager lock during graceful shutdown gets [cleaned up here](https://github.com/goosemania/pulp/blob/b8abcc929cc3cc4d0a0c2a531d89f2d17480d24f/tasking/pulp/tasking/base.py#L156-L157). Note that the celerybeat code also calls that from [here](https://github.com/goosemania/pulp/blob/b8abcc929cc3cc4d0a0c2a531d89f2d17480d24f/tasking/pulp/tasking/scheduler.py#L286). We should add the celerybeat lock release next to the resource manager lock release in the same place.

Also, as an aside, I think the resource manager lock removal needs to be moved outside of [the if statement](https://github.com/goosemania/pulp/blob/b8abcc929cc3cc4d0a0c2a531d89f2d17480d24f/tasking/pulp/tasking/base.py#L156) that current guards it.

I think our non-graceful failover case is broken so I'm going to look into some recommendations there.

One question I have is what is the purpose of [this field](https://github.com/goosemania/pulp/blob/b8abcc929cc3cc4d0a0c2a531d89f2d17480d24f/app/pulp/app/models/task.py#L82)? I think it can be removed.